packages/trader/src/Stores/client-store.js

```diff
@@ -594,8 +592,17 @@ export default class ClientStore extends BaseStore {
                 };
                 await this.init(new_user_login);
 
-                // TODO: clean-refresh-trade: Find a way to do this without re-preparing the trade store
-                this.root_store.modules.trade.onMount();
+                // Refresh trade-store currency and proposal before requesting new proposal upon login
+                runInAction(async() => {
```

Comment:
Why can't this be a method of the Trade store?

Reply:
I can move it to `trade-store` if you want but I doubt it will be used anywhere else but here.

Reply:
@zaki-binary That would be great, thank you. I think it would be better to not imperatively do these in the `onSignup()`, and even better for it to not be in Client store as that has little relevancy.

Reply:
@mahdipourziaei-binary , sure. I'll move the method to `trade-store` then.